to-do-list/src/components/Task.js

```diff
@@ -0,0 +1,41 @@
+import { useState } from "react";
+import Button from "./Button"
+
+function Task(props) {
+
+    const { task, date, status, completeTask, curDate } = props
+
+    const dueDate = new Date(date) 
+
+    let difference = dueDate - curDate; //subtract Due Date with Current Date
+
+    const differenceDays = Math.floor(difference / (1000 * 60 * 60 * 24)) //Days
+
+    const differenceTime = Math.floor((difference / (1000 * 60 * 60)) % 24)//Time
+
+    let lateDue = curDate - dueDate; 
+
+    const lateDays = Math.floor(lateDue / (1000 * 60 * 60 * 24))
+
+    const lateTime = Math.floor((lateDue / (1000 * 60 * 60)) % 24)
+
+    const deleteTask = function () {
+        props.deleteTask(props.id)
+    }
```

Comment:
Since we already know the id of this task when we render this component (up at the TaskList component level), we could actually define this function up there and send this down as a prop.

So up at TaskList, render it with `deleteTask={() => deleteTask(taskItem.id)}`. That way in this component you don't have to define this function, you can just make the onClick for the button (below) `onClick={props.deleteTask}`.

In general don't pass down props unless you have to!